src/utils/form/addField.ts

```diff
@@ -1,14 +1,35 @@
 import { getQuestion } from './getQuestion';
 
 /**
- * Check if the structure is correct and add a new field at the beginning. Use a passed structure to fetch the correct question.
+ * Check if the structure is correct and add a new field. Use a passed structure to fetch the correct question.
+ * The method tries to put the new question at the same place than in the template.
  * Function by induction.
  * @param structure structure of the form to edit
- * @param name name of the fied to search for
+ * @param name name of the field to search for
  * @param template structure of the core template
  */
 export const addField = (structure: any, name: string, template: any): void => {
-  if (structure.pages && structure.pages.length > 0 && structure.pages[0].elements) {
-    structure.pages[0].elements.unshift(getQuestion(template, name));
-  }
+  let prevQuestion = '';
+  let isFirstQuestion = false;
+  let pageIndex = 0;
+
+  // Find in the template the previous question name
+  template.pages.forEach((page, pIndex) => page.elements.findIndex((el, qIndex) => {
```

Comment:
@Hugo-Plissonneau if fact it is not working if the question is in another element, like a panel